### DISCUSSION TITLE
Windows Installer

### DISCUSSION THREAD

**Discussion by lainz** [mse-org/mseide-msegui#58] [General]
Hi, it will be nice to have some Inno Setup script, to install and configure FPC with mseide-msegui.
Say FPC 3.2.0 and the latest IDE ready to use, already configured for the folder the user selects.

**Comment by fredvs** [comment]
Hello.

There is already a "mse Inno Setup" installer.

Maybe you could use this one.
Take a look at:
https://github.com/mse-org/mseuniverse/tree/master/tools/mse-installer

There are 2 projects, one looks like Inno Setup and, of course, it is multi os.

**Comment by fredvs** [comment]
Roland has proposed to do it some time ago but I dont know if it is already ready.

**Comment by lainz** [comment]
Nice to know, I will try that

**Comment by rchastain** [comment]
> Roland has proposed to do it some time ago but I dont know if it is already ready.

Hello! Indeed I started to work on that but never finished. Now I don't use Windows at all.

**Comment by lainz** [comment]
Hi and do you remember how it was like?

For example I can see ideU has only an ini file as configuration, that for sure can be accessed and modified by Inno Setup at installation time to setup the paths.

The FPC config file is a thing I never touched, what about that?

**Comment by fredvs** [comment]
Hello Leandro.

> Hi and do you remember how it was like?

Are you talking about a installer for MSEide (the binary of MSEide) ?
If so, in the release there are only 2 files inside the zip file : mseide(.exe) and readme.txt

MSEide is 'out-of-the-box', you only need to unzip it and run mseide(.exe).
No ini file is needed, all the infos of your last session is stored inside a 'sta' (status) file.

So, to resume, there is in fact no need for a installer.

But maybe people like it for Windows, but the work of the installer will be very short, basically only unzip the zip file somewhere.

But yes also the installer would be good to install shorcut or some Windows things.

**Comment by lainz** [comment]
Ok. So is in fact not needed.
Ok thanks.

**Comment by fredvs** [comment]
Maybe it could be good for beginners on Windows.

Using the **Inno Setup** well known interface to help people to choose a directory where to unzip the file.
And, the most important, imho, at end of setup, a message that says:

_Dont forget to setup the macros, with menu "`Settings`" - "`Configure MSEide`".
"Have lot of fun!"_

**Comment by rchastain** [comment]
It's true that an installer is not really needed ; but it would be a good exercise to do it, and it could make new users come to MSEide.

IMHO the installer should do automatically the configuration. We had a discussion about that [here](http://mseide-msegui-talk.13964.n8.nabble.com/MSEide-MSEgui-talk-Procedure-to-create-the-configuration-file-of-MSEide-td2257.html).

For me I am still interested in doing an installer for Linux. I began to work on that. It isn't finished, but I still think of it.

**Comment by fredvs** [comment]
> but it would be a good exercise to do it

Imho, it would be a perfect exercise to discover how mseide works.
Even more, if you use the "mse Inno Setup clone" and customize it, you will directly become a msegui guru.

Fre;D

**Comment by lainz** [comment]
To make it simple, it can use the FPC installer inside the inno setup script, it is possible to do so. 

If the user choose to install also FPC, it can be installed automatically, if not, the user must provide the path or say that's already installed.

It must install everything required, so you can just create a new project and compile without touching any configuration, like is done when you install Lazarus.

**Comment by fredvs** [comment]
Ping ?

**Comment by lainz** [comment]
Pong :)

Hi. I'm not doing another thing than working lately. That's a good thing
BTW. I'm happy with that. So no open source right now.

**Comment by fredvs** [comment]
Hola Lainz.

Nice that you have work!

The Ping was mostly to ask if somebody is still interested to do it.
If no, maybe I will jump into it, (if I find time).

**Reply by lainz** [reply]
Hi. A lot of time since this.

Seems that if an installer is not needed then is better to don't have one.

I'm being used to don't have installers in some apps like cudatext that's built with fpc as well.

And some windows command line apps...
